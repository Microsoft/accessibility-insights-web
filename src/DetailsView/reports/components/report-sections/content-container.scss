// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import './common-mixins.scss';

.content-container {
    @include contentSize();
    margin-top: 24px;

    h2 {
        margin: 0px;
        font-size: 17px;
        line-height: 24px;
    }

    @media only screen and (max-width: 1000px) {
        margin: 24px 16px 0 16px;
    }
}
