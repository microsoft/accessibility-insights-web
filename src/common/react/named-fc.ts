// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

export type ReactFCWithDisplayName<P = {}> = React.FC<React.PropsWithChildren<P>> & {
    displayName: string;
};

export function NamedFC<P = {}>(
    displayName: string,
    component: React.FC<React.PropsWithChildren<P>>,
): ReactFCWithDisplayName<P> {
    component.displayName = displayName;

    return component as ReactFCWithDisplayName<P>;
}
