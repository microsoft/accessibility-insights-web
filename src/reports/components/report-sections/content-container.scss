// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import './common-mixins.scss';

.outer-container {
    box-shadow: 0 1px 0 rgb(0 0 0 / 8%);

    .content-container {
        @include content-size;

        margin-top: 24px;

        h2 {
            margin: 0;
            font-size: 17px;
            line-height: 24px;
        }

        @media only screen and (width <= 1000px) {
            margin-top: 24px;
            margin-right: 16px;
            margin-left: 16px;
        }
    }
}
