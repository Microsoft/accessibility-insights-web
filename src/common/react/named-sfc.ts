// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

export type ReactSFCWithDisplayName<P = {}> = React.SFC<P> & { displayName: string };

export function NamedSFC<P = {}>(displayName: string, component: React.SFC<P>): ReactSFCWithDisplayName<P> {
    component.displayName = displayName;

    return component as ReactSFCWithDisplayName<P>;
}
