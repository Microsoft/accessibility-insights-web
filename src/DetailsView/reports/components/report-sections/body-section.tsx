// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';
import { NamedSFC } from 'common/react/named-sfc';

export const BodySection = NamedSFC('BodySection', ({ children }) => {
    return <body>{children}</body>;
});
