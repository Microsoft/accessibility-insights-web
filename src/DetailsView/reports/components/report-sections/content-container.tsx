// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';
import { NamedSFC } from '../../../../common/react/named-sfc';

export const ContentContainer = NamedSFC('ContentSection', ({ children }) => {
    return <div className="content-section">{children}</div>;
});
