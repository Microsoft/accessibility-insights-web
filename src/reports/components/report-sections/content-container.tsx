// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedSFC } from 'common/react/named-sfc';
import * as React from 'react';

export const ContentContainer = NamedSFC('ContentSection', ({ children }) => {
    return (
        <main className="outer-container">
            <div className="content-container">{children}</div>
        </main>
    );
});
