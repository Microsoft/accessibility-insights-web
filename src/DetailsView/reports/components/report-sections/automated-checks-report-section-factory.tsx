// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { NamedSFC } from '../../../../common/react/named-sfc';
import { BodySection } from './body-section';
import { ContentContainer } from './content-container';
import { DetailsSection } from './details-section';
import { FooterSection } from './footer-section';
import { HeaderSection } from './header-section';
import { ReportSectionFactory } from './report-section-factory';
import { TitleSection } from './title-section';

const createBasicComponent = (name: string) => {
    return NamedSFC(name, () => {
        return <div id={name} />;
    });
};

const createWrappingComponent = (name: string) => {
    return NamedSFC(name, ({ children }) => {
        return <div id={name}>{children}</div>;
    });
};

const Summary = createBasicComponent('summary-section');
const ResultSection = createWrappingComponent('result-section');
const FailedInstances = createBasicComponent('failed-instances-section');
const PassedChecks = createBasicComponent('passed-checks-section');
const NotApplicableChecks = createBasicComponent('not-applicable-checks-section');

// TODO most of this sections are dummy sections, the point is to replace them as we develop them
export const AutomatedChecksReportSectionFactory: ReportSectionFactory = {
    BodySection,
    ContentContainer,
    Header: HeaderSection,
    Title: TitleSection,
    Summary,
    Details: DetailsSection,
    ResultSection,
    FailedInstances,
    PassedChecks,
    NotApplicableChecks,
    Footer: FooterSection,
};
