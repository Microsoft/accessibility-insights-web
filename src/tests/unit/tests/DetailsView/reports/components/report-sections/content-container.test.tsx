// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import * as React from 'react';

import { ContentContainer } from '../../../../../../../DetailsView/reports/components/report-sections/content-container';

describe('ContentContainer', () => {
    it('renders', () => {
        const children: JSX.Element[] = [<div>1</div>, <div id="2">2</div>];

        const wrapped = shallow(<ContentContainer>{children}</ContentContainer>);

        expect(wrapped.getElement()).toMatchSnapshot();
    });
});
