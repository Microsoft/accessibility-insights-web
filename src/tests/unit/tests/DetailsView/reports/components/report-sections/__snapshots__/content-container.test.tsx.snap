// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentContainer renders 1`] = `
<div
  className="content-container"
>
  <div>
    1
  </div>
  <div
    id="2"
  >
    2
  </div>
</div>
`;
